DGC-all-schemas-combined.json

```diff
@@ -0,0 +1,390 @@
+          "type": "string"
+        },
+        "ma": {
+          "description": "Test Manufacturer",
+          "$ref": "#/$defs/test-manf"
+        },
+        "sc": {
+          "description": "Date/Time of Sample Collection",
+          "type": "string",
+          "format": "date-time"
+        },
+        "dr": {
+          "description": "Date/Time of Test Result",
+          "type": "string",
+          "format": "date-time"
+        },
+        "tr": {
+          "description": "Test Result",
+          "$ref": "#/$defs/test-result"
+        },
+        "tc": {
+          "description": "Testing Centre",
+          "type": "string",
+          "maxLength": 50
+        },
+        "co": {
+          "description": "Country of Test",
+          "$ref": "#/$defs/country_vt"
+        },
+        "is": {
+          "description": "Certificate Issuer",
+          "$ref": "#/$defs/issuer"
+        },
+        "ci": {
+          "description": "Unique Certificate Identifier, UVCI",
+          "$ref": "#/$defs/certificate_id"
+        }
+      }
+    },
+    "recovery_entry": {
+      "description": "Recovery Entry",
+      "required": [
+        "tg",
+        "fr",
+        "co",
+        "is",
+        "df",
+        "du",
+        "ci"
+      ],
+      "type": "object",
+      "properties": {
+        "tg": {
+          "$ref": "#/$defs/disease-agent-targeted"
+        },
+        "fr": {
+          "description": "ISO 8601 Date of First Positive Test Result",
+          "type": "string",
+          "format": "date"
+        },
+        "co": {
+          "description": "Country of Test",
+          "$ref": "#/$defs/country_vt"
+        },
+        "is": {
+          "description": "Certificate Issuer",
+          "$ref": "#/$defs/issuer"
+        },
+        "df": {
+          "description": "ISO 8601 Date: Certificate Valid From",
+          "type": "string",
+          "format": "date-time"
+        },
+        "du": {
+          "description": "Certificate Valid Until",
+          "type": "string",
+          "format": "date-time"
+        },
+        "ci": {
+          "description": "Unique Certificate Identifier, UVCI",
+          "$ref": "#/$defs/certificate_id"
+        }
+      }
+    },
+    "disease-agent-targeted": {
+      "description": "EU eHealthNetwork: Value Sets for Digital Green Certificates. version 1.0, 2021-04-16, section 2.1",
+      "type": "string",
+      "required": [ "code" ],
+      "enum": [ "840539006" ]
+    },
+    "vaccine-prophylaxis": {
+      "description": "EU eHealthNetwork: Value Sets for Digital Green Certificates. version 1.0, 2021-04-16, section 2.2",
+      "type": "string",
+      "enum": [
+        "1119305005",
+        "1119349007",
+        "J07BX03"
+      ]
+    },
+    "vaccine-medicinal-product": {
+      "description": "EU eHealthNetwork: Value Sets for Digital Green Certificates. version 1.0, 2021-04-16, section 2.3",
+      "type": "string",
+        "enum": [
```

Comment:
Using enums means that a new schema version will be needed for every little addition. Which means that many apps needs to be re-built ...

We could specify allowed values in another way ...

Reply:
So what do we do ? leave enum and ask an any value  ? leave enum and document that you should NOT use this for validation in the field ? Kill enum and say 'free form'

Reply:
keep enum ->  value sets like this are already separated into DGC.ValuesSets.schema.json

Reply:
But even if the definitions are placed in a separate JSON-file, how will that help people generating code from the schema? It will still end up as enums in their code.